sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/ScalaReflectionSuite.scala

```diff
@@ -237,4 +241,45 @@ class ScalaReflectionSuite extends SparkFunSuite {
     assert(anyTypes.forall(!_.isPrimitive))
     assert(anyTypes === Seq(classOf[java.lang.Object], classOf[java.lang.Object]))
   }
+
+  private def testThreadSafetyFor(name: String)(exec: () => Any) = {
+    test(s"thread safety of ${name}") {
+      for (_ <- 0 until 100) {
```

Comment:
Yes I understand repeating it 100 times, but your idiom for iterating n times is different on two nearby lines.

I actually meant the opposite -- remove the helper method but run `test`s in a loop still -- but leave it.

My question was more like, does the extra locking cause trouble in 2.11? I know you said 2.11 doesn't need this change per se.